omegaconf/basecontainer.py

```diff
@@ -50,12 +50,9 @@ def _resolve_with_default(
     ) -> Any:
         """returns the value with the specified key, like obj.key and obj['key']"""
 
-        def is_mandatory_missing(val: Any) -> bool:
-            return bool(get_value_kind(val) == ValueKind.MANDATORY_MISSING)
-
         val = _get_value(value)
         has_default = default_value is not DEFAULT_VALUE_MARKER
-        if has_default and (val is None or is_mandatory_missing(val)):
+        if has_default and (val is None or _is_missing_value(val)):
```

Comment:
Can this one be `_is_missing_literal()`?

Reply:
Tried. no.
since _get_value does not unwrap container nodes, _is_missing_literal rejects the input in some scenarios.

Reply:
I was confused so I tried it, and now I understand (it's also related to my previous comment).

The problem is because of the `assert not isinstance(value, Node)` in `_is_missing_literal`. If you remove it, then what I suggested makes sense, but it doesn't otherwise. I'm fine with either option, then.

Reply:
gotcha.
I was trying to draw a very clear line between checking if a node is missing versus checking if an actual value is missing.
let's keep the assert.

Reply:
Ok sounds good, resolving